From 37ec2316f4ab17f79173e8ea675242b1e88720ba Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Thu, 16 Jan 2014 20:21:27 +0000
Subject: [PATCH 0515/1076] MANTIS 0030448 extension; SVC Application
 activation via specialized barcode

platform.idl
V1.4

added method dataReceived, to allow device to forward input to platform


git-svn-id: svn://localhost/SelfServiceCommon/trunk@640 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/idl/Massai/platform.idl            | 44 +++++++++++--------
 1 file changed, 26 insertions(+), 18 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/platform.idl b/SelfServiceCommon/Massai/idl/Massai/platform.idl
index 51544a89..361349b0 100644
--- a/SelfServiceCommon/Massai/idl/Massai/platform.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/platform.idl
@@ -28,7 +28,7 @@
 
 module PlatformManagement
 {
-    const string interface_version = "1.3";
+    const string interface_version = "1.4";
 
     /** Data type for text lists. */
 
@@ -181,7 +181,7 @@ module PlatformManagement
      */
 
 
-	enum LoggingClass
+    enum LoggingClass
     {
       LC_APPLICATION,
       LC_PRINTER,
@@ -199,6 +199,7 @@ module PlatformManagement
       LC_NETWORK,
       LC_UNDEFINED
     };
+    
     struct LoggingComponentData 
     {
       string componentRef;      // identify virtual component by virtualComponentRef 
@@ -208,29 +209,29 @@ module PlatformManagement
                                 // - LC_CAPTURE and LC_DISPENSER may be responsible for multiple units                                
       LoggingClass unitClass;   // class of component 
                                 //   LC_BIN* not used in LoggingComponentData
-      string moduleName;	// needed for GAC_MODULENAME
+      string moduleName;    // needed for GAC_MODULENAME
+    };
+    
+    struct LoggingCounterData
+    {
+      string realComponentName;
+      LoggingClass unitClass;
+      long bin;                 //bin number
+      long mediaStatusTypeValue;//int value of mediaStatusType
+      long couponsproduced;
+      long numberOfDocs;
+      long paperType;           //org.iata.cuss.Characteristics.MediaOutputPackage;
+      string paperClass;        //com.materna.massai.DeviceComponents.characteristics.types
     };
-	
-	struct LoggingCounterData
-	{
-	  string realComponentName;
-	  LoggingClass unitClass;
-	  long bin;                 //bin number
-	  long mediaStatusTypeValue;//int value of mediaStatusType
-	  long couponsproduced;
-	  long numberOfDocs;
-	  long paperType;           //org.iata.cuss.Characteristics.MediaOutputPackage;
-	  string paperClass;        //com.materna.massai.DeviceComponents.characteristics.types
-	};
     
     interface PlatformIF
     {
 
-	typedef sequence<LoggingComponentData> LoggingComponentDataList;   
-    	void initLogging(in string realComponentName, in LoggingComponentDataList init);
+    typedef sequence<LoggingComponentData> LoggingComponentDataList;   
+        void initLogging(in string realComponentName, in LoggingComponentDataList init);
         void logCounter(in LoggingCounterData lcd);
         void logCounterFailure(in string realComponentName, in LoggingClass unitClass, in long numberOfFailed);
-		
+        
         /** Checks the status of the token passed by the application.
          *  @param token - The application token
          *  @return the token status of the application token.
@@ -332,6 +333,13 @@ module PlatformManagement
          *  @return ComponentServiceIF
          **/
         ComponentServiceIF getComponentService(in string realDeviceName);
+
+
+        /** a device has received data
+         *  @param componentType - type of the component that received the data; e.g.: BCR for barcode reader
+         *  @param data - data as defined in cuss types.idl (msgDataType, aeaDataType, svgDataType, ...)
+         **/
+        void dataReceived(in string componentType, in any data);
     };
 };
 
-- 
2.41.0.windows.1

